### PR TITLE
Changes to the test function

### LINKED ISSUE #657: on_train_end callback is not called if training is stopped early
```python
            # early stopping
            met_min_epochs = epoch >= self.min_epochs - 1
            if self.enable_early_stop and (met_min_epochs or self.fast_dev_run):
                should_stop = self.early_stop_callback.on_epoch_end(epoch=epoch,
                                                                    logs=self.callback_metrics)
                # stop training
                stop = should_stop and met_min_epochs
                if stop:
                    self.main_progress_bar.close()
                    return

        self.main_progress_bar.close()

        model.on_train_end()
```

### PR DESCRIPTION
**Pull request by HarshSharma12**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes #657. Overrides PR #726 

> Updates as per the comments (#723) and changes to the test function

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by williamFalcon** [comment]
@HarshSharma12 @Borda is this still active?

File: tests/utils.py [hunk 1]
```diff
@@ -65,6 +65,10 @@ def run_model_test(trainer_options, model, on_gpu=True):
     # correct result and ok accuracy
     assert result == 1, 'amp + ddp model failed to complete'
 
+    if early_stop:
+        assert trainer.current_epoch >= trainer.min_epochs, 'amp + ddp model failed to complete'
```
Comment (Borda) [inline tests/utils.py]: rather make is raise an error

**Comment by Borda** [comment]
@HarshSharma12 may you rebase, pls?

**Comment by ethanwharris** [comment]
@Borda @williamFalcon @HarshSharma12 I had a look at rebasing this - the issue seams to have been fixed by #856 - see [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/6b667b1237f8ed74db920a2c1c49d55ca1d2b7e5/pytorch_lightning/trainer/training_loop.py#L630)

I vote for closing this PR and the associated issue

**Comment by Borda** [comment]
Ok, closing this... anyway very thanks @HarshSharma12 for your help and looking forward to you future help :robot:

**Pull request closed without merging**